### PR TITLE
feat(sdk): adding cloudfront oac to tf-aws `cloud.website`

### LINKED ISSUE #3538: Website: Cloudfront should authorize with Origin Access Identity against the Website Bucket
### Feature Spec

Now the Website SDK is using Cloudfront [origin access identity](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_origin_access_identity) to authenticate against the S3 bucket which holds the assets for the website. 

Up until now, the website bucket had to be made public for everyone, which is unnecessary in this case and could lead to nagging security warnings in security scanning tools.

### Use Cases

- follow best practices for setting up AWS resources
- Avoid unneccessary `*` principal
- Avoid security warnings from infra security tools (e.g. AWS Config) or even teams

 

### Implementation Notes

Stumbled upon this while troubleshooting https://github.com/winglang/wing/pull/3536 - which is a short term fix for the existing implementation. If we had this one here implemented, the other fix wouldn't be necessary for the Website. It still makes sense for the cases where a user actually want's to have just a public bucket.

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by garysassano**
Closes #3538

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/target-tf-aws/website.ts [hunk 1]
```diff
@@ -35,6 +37,18 @@ export class Website extends cloud.Website {
 
     this.uploadFiles(this.path);
 
+    // create a cloudfront oac
+    const cloudfrontOac = new CloudfrontOriginAccessControl(
+      this,
+      "CloudfrontOac",
+      {
+        name: "cloudfront-oac",
```
Comment (eladb) [inline libs/wingsdk/src/target-tf-aws/website.ts:45]: Can we make the bucket non-public?

Comment (garysassano) [reply libs/wingsdk/src/target-tf-aws/website.ts:45]: Isn't the S3 bucket already private by default?

**Comment by tsuf239** [comment]
image
Tried to run `examples/tests/sdk_tests/website/website.w` from your branch and got the error above, can you please take a look?

**Comment by garysassano** [comment]
@tsuf239 Thanks for reporting. Could you please try again?

**Comment by tsuf239** [comment]
yes, it works now: 
image

**Review by tsuf239 — APPROVED** [review]

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**